### ISSUE TITLE
Want to delete the key "tm-404" by default

### ISSUE THREAD

**Issue by morix1500** [fastly/WordPress-Plugin#36]
Hello!

WordPress has a feature called Schedule Posts.
Accessing the URL of that page before the date and time specified in the reservation post will result in 404.
Purge will be requested at the time of reservation posting, but the posting article page cache does not disappear.
The reason was that the Surrogate-Key on the article page would be "tm-404".
"tm-404" is not the key to be deleted by default.

I'd like to deal with adding "tm-404" to the following array, but are there other good ideas?
https://github.com/fastly/WordPress-Plugin/blob/master/src/classes/related-surrogate-keys.php#L76

**Comment by vvuksan** [comment]
We'll look into it.

**Comment by tnikcevs** [comment]
Hello,

404 page is not what should be purged, as this is what you end up on after not finding content. The issue is that post is not purged when WP tries to publish it automatically.

Can you let me know if you have this issue only with posts that have "Missed schedule" in post listing if automatic scheduling failed, or also on the ones that where published automatically with success?

Regards,
Tomislav

**Comment by morix1500** [comment]
>404 page is not what should be purged, as this is what you end up on after not finding content. The issue is that post is not purged when WP tries to publish it automatically.

That's right. Thank you for your advice.

It does not mean that automatic posting failed. 
If you access the page before the publication time, the cache of that page will not be deleted.
Although this may be a rare case.

Normal Surrogate-Key(If you did not access the page before publishing):

```
https://example.global.ssl.fastly.net/2017/11/24/post1/
Surrogate-Key:p-129 tm-single t-1 a-1
```

Surrogate-key when accessing page before publication:

```
https://example.global.ssl.fastly.net/2017/11/24/post2/
Surrogate-Key:tm-404
```

Is it possible to include the page ID in the surrogate key of the page before publishing?

**Comment by tnikcevs** [comment]
Hello,

You are right regarding tm-404 key, the problem was that only this key was set if someone attempted to access the url before it existed, which means that when post was published, tm-404 was not purged due to not being connected to new post via surrogate keys. We have added tm-404 on always purge list so this should fix the issue.
The fix should be available in next version as soon as it is out.

Regards,
Tomislav

**Comment by morix1500** [comment]
Thank you for fixing it!

**Comment by vvuksan** [comment]
Fix for this issue is included in #37 and released as version 1.2.4 of the Wordpress plugin.
